test/config/test_configurable.rb

```diff
@@ -164,10 +148,115 @@ class Example6
     config_param :obj1, :hash, default: {}
     config_param :obj2, :array, default: []
   end
+
+  module Overwrite
+    class Base
+      include Fluent::Configurable
+
+      config_param :name, :string, alias: :fullname
+      config_param :bool, :bool, alias: :flag
+      config_section :detail, required: false, multi: false, alias: "information" do
+        config_param :address, :string, default: "x"
+      end
+    end
+
+    class Required < Base
+      config_section :detail, required: true do
+        config_param :address, :string, default: "x"
+      end
+    end
+
+    class Multi < Base
+      config_section :detail, required: true do
+        config_param :address, :string, default: "x"
+      end
+    end
+
+    class Alias < Base
+      config_section :detail, alias: "information2" do
+        config_param :address, :string, default: "x"
+      end
+    end
+
+    class DefaultOptions < Base
+      config_section :detail do
+        config_param :address, :string, default: "x"
+      end
+    end
+
+    class DetailAddressDefault < Base
+      config_section :detail do
+        config_param :address, :string, default: "y"
+      end
+    end
+
+    class AddParam < Base
```

Comment:
in_tcp and in_udp overwrites default value of base class.

* https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/in_tcp.rb#L25
* https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/in_udp.rb#L23

Do you want to raise exception in such case?